adminSiteServer/apiRouter.ts

```diff
@@ -3051,6 +3059,169 @@ postRouteWithRWTransaction(
     }
 )
 
+getRouteNonIdempotentWithRWTransaction(
+    apiRouter,
+    "/images.json",
+    async (_, res, trx) => {
+        try {
+            const images = await db.getCloudflareImages(trx)
+            res.set("Cache-Control", "no-store")
+            res.send({ images })
+        } catch (error) {
+            console.error("Error fetching images", error)
+            res.status(500).json({
+                error: { message: String(error), status: 500 },
+            })
+        }
+    }
+)
+
+postRouteWithRWTransaction(apiRouter, "/image", async (req, res, trx) => {
+    const { filename, type, content } = req.body
+    if (!filename || !type || !content) {
+        throw new JsonError("Missing required fields", 400)
+    }
+    if (
+        typeof filename !== "string" ||
+        typeof type !== "string" ||
+        typeof content !== "string"
+    ) {
+        throw new JsonError("Invalid field types", 400)
+    }
+
+    // Strip the data URL prefix
+    const stripped = content.slice(content.indexOf(",") + 1)
+    const asBuffer = Buffer.from(stripped, "base64")
+    const asBlob = new Blob([asBuffer], { type: type })
+    const dimensions = await sharp(asBuffer)
+        .metadata()
+        .then(({ width, height }) => ({ width, height }))
```

Comment:
Hmm, maybe I'm still missing something, but what I meant is keeping the same data flow, i.e. uploading the image to the server before we send it to CF, just using `mutlipart/form-data` instead of base64.

Reply:
Hey! So I added a comment that addressed this in the [next PR that adds image reupload functionality](https://github.com/owid/owid-grapher/pull/4252)

In brief, `multipart/form-data` required wiring up the `multer` express middleware and adding a new parameter to our router util functions. It seemed like a heavier amount of complexity to add compared to a small amount of in situ blob manipulation, so I went with the lighter footprint option.

If I thought we were going to be adding more API routes that would need to be able to handle uploads, I would have gone with the `multer` approach, but I think this will probably be a one-off. 🙂

Reply:
OK, let's leave it as blob for now, then. I think we should get rid of the `FunctionalRouter`, but that's a different story.